File: src/game/tools/rpc/output/wrap/QuestDataListWrap.go
package wraps

import (
	rpc "tipcat.com/common/rpc/simple"
)

type QuestDataListWrap struct {
	
	QuestList []*QuestDataWrap // 关卡列表
	rpc.Wrapper
}

func (w *QuestDataListWrap)Decode(pck *rpc.Packet) rpc.Wrapper {
	
	w.QuestList = make([]*QuestDataWrap, int(pck.PopInt16()))
	for i := 0; i < len(w.QuestList); i++ {
		w.QuestList[i] = new(QuestDataWrap)
		w.QuestList[i].Decode(pck)
		
	}
	
	return w
}

func (w *QuestDataListWrap)Encode(pck *rpc.Packet) {
	
	if w.QuestList == nil {
		pck.PutInt16(0)
	} else {
    	pck.PutInt16(int16(len(w.QuestList)))
    	for i := 0; i < len(w.QuestList); i++ {
    		w.QuestList[i].Encode(pck);
			
    	}
    }
	
}

File: src/game/tools/rpc/output/wrap/QuestDataListWrap.cs
using System;
using Common.Net.Simple;

namespace Game 
{
	public class QuestDataListWrap : IWrapper
	{	
		public QuestDataWrap[] QuestList; // 关卡列表
		
		public void Decode(Packet pck)
		{	
			QuestList = new QuestDataWrap[pck.GetShort()];
			for (int i = 0; i < QuestList.Length; i++)
			{
				QuestList[i] = new QuestDataWrap();
				QuestList[i].Decode(pck);
				
			}
			
		}
		
        public void Encode(Packet pck)
        {	
        	if (QuestList == null) pck.PutShort((short)0); 
        	else
        	{
	        	pck.PutShort((short)QuestList.Length);
	        	for(int i = 0; i < QuestList.Length; i++)
	        	{
	        		QuestList[i].Encode(pck);
					
	        	}
	        }
        	
        }
	}
}